lib/git_worktree.rb

```diff
@@ -258,6 +258,22 @@ def checkout
     end
   end
 
+  # Ping the git repository endpoint to verify that the network connection is still up.
+  #
+  def check_connection(url)
+    require 'net/ping/http'
+
+    # URI library cannot handle git urls, so just convert it to a standard url.
+    url = url.sub(':', '/').sub('git@', 'https://www.') if url.start_with?('git@')
```

Comment:
Is the `www.` always valid for a URL?  For say, `github.com`, I would guess it is, but probably not for every `git` host out there and is probably dependent on the domain name configuration for a given host.

It probably makes sense to just omit the addition of `www.` addition here just to be safe.

Reply:
private gitlab most likely won't (ours doesn't)

Reply:
Without the 'www' it's a 301 redirect:
```
dberger:~/Dev >curl -Is https://www.github.com/djberg96/sys-uname | grep HTTP | cut -d ' ' -f2
301
dberger:~/Dev >curl -Is https://github.com/djberg96/sys-uname | grep HTTP | cut -d ' ' -f2 
200
```
I thought the net-ping library followed them, but apparently not, or it's just not working.

Reply:
Actually, do I have it backwards? It works on a redirect? Hm, lemme check.

Reply:
Nevermind this, we're just doing an external ping on the hostname now.